types/test-types.tsx

```diff
@@ -0,0 +1,249 @@
+    onFocus: () => {},
+    onReady: () => {},
+    onSubmit: () => {}
+  };
+
+  const cardElementProps = {
+    ...commonElementProps,
+    inputType: 'mobileSelect',
+    displayIcon: true,
+    style: {
+      fontSize: '1em',
+      placeholder: {
+        color: 'gray !important',
+        fontWeight: 'bold',
+        content: {
+          number: 'Card number',
+          cvv: 'CVC'
+        }
+      },
+      invalid: {
+        fontColor: 'red'
+      }
+    },
+    onChange: (e: CardElementChangeEvent) => {
+      // $ExpectType boolean
+      e.valid;
+      // $ExpectType string
+      e.firstSix;
+      // $ExpectType string
+      e.lastFour;
+      // $ExpectType string
+      e.brand;
+      // $ExpectType boolean
+      e.empty;
+      // $ExpectType boolean
+      e.focus;
+    }
+  };
+
+  const elementProps = {
+    ...commonElementProps,
+    format: true,
+    inputType: 'text',
+    tabIndex: '1',
+    style: {
+      invalid: {},
+      padding: '10px',
+      placeholder: {
+        color: 'red',
+        content: 'content'
+      }
+    },
+    onChange: (e: IndividualElementChangeEvent) => {
+      // $ExpectType boolean
+      e.empty;
+      // $ExpectType number
+      e.length;
+      // $ExpectType boolean
+      e.valid;
+      // $ExpectType boolean
+      e.focus;
+    }
+  };
+
+  const CardNumberProps = {
+    ...elementProps,
+    onChange: (e: CardNumberElementChangeEvent) => {
+      // $ExpectType boolean
+      e.valid;
+      // $ExpectType string
+      e.firstSix;
+      // $ExpectType string
+      e.lastFour;
+      // $ExpectType string
+      e.brand;
+      // $ExpectType number
+      e.length;
+      // $ExpectType boolean
+      e.empty;
+      // $ExpectType boolean
+      e.focus;
+    }
+  };
+
+  const checkoutPricingInput: UseCheckoutPricingInput = {
+    subscriptions: [
+      {
+        plan: 'my-plan',
+        quantity: 3,
+        addons: [{ code: 'item code', quantity: 5 }]
+      }
+    ],
+    adjustments: [
+      {
+        itemCode: "item-code",
+        quantity: 3,
+        currency: "USD"
+      },
+      {
+        id: 'id',
+        amount: 10,
+        quantity: 3,
+        currency: "USD",
+        taxExempt: false,
+        taxCode: "tax-code",
+      }
+    ],
+    currency: "USD",
+    address: {
+      first_name: "first",
+      last_name: "last",
+      address1: "address 1",
+      address2: "address 2",
+      city: "city",
+      state: "state",
+      postal_code: "postal code",
+      country: "US",
+      phone: "phone",
+      vat_number: "vat number"
+    },
+    giftCard: {
+      code: "gift-card-code"
+    }
+  };
+
+  const [{price, loading}, setPricing] = useCheckoutPricing(checkoutPricingInput);
+
+  setPricing(checkoutPricingInput);
+  // $ExpectError
+  setPricing();
+  // $ExpectError
+  setPricing({ ...checkoutPricingInput, fake: "fake" });
+
+  // $ExpectType boolean
+  loading;
+  // $ExpectType string
+  price.now.items[0].addons;
```

Comment:
There might be a better way to make these assertions. We're already testing this time in [recurly-js](https://github.com/recurly/recurly-js/blob/master/test/types/pricing/checkout.ts#L103-L164)

Reply:
Seems reasonable since the cause differs

Reply:
That's what I was thinking, but I wasn't sure